group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1307,13 +1307,14 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
             }
         }
 
+        // The subscription metadata is updated in two cases:
+        // 1) The member has updated its subscriptions;
+        // 2) The refresh deadline has been reached.
+        Map<String, Integer> subscribedTopicNamesMap = group.subscribedTopicNames();
```

Comment:
While here, could we move those two [lines](https://github.com/apache/kafka/pull/15785/files#diff-00f0f81cf13e66781777d94f7d2e68a581663385c37e98792507f2294c91bb09R1275) to here. It makes sense to keep all those together.